newIDE/app/src/EventsSheet/EventsTree/DropContainer.js

```diff
@@ -0,0 +1,295 @@
+  'bottom-bottom-left': {
+    left: isDraggedNodeChild ? '0px' : '10px',
+    top: '100%',
+    height: draggedNodeHeight,
+    width: indentWidth,
+  },
+  'bottom-bottom-right': {
+    left: isDraggedNodeChild ? `${indentWidth + 10}px` : `${indentWidth}px`,
+    right: '0px',
+    top: '100%',
+    height: draggedNodeHeight,
+  },
+});
+
+const getIndicatorPositionStyles = (indentWidth: number) => ({
+  bottom: { left: '0px', right: '0px', bottom: '-4px' },
+  'bottom-right': { left: `${indentWidth}px`, right: '0px', bottom: '-4px' },
+  top: { left: '0px', right: '0px', top: '-4px' },
+});
+
+type DropTargetContainerStyle = {|
+  left?: string,
+  right?: string,
+  top?: string,
+  bottom?: string,
+  width?: number,
+  height?: number,
+|};
+
+function DropTargetContainer({
+  DnDComponent,
+  canDrop,
+  onDrop,
+  style,
+}: {|
+  DnDComponent: any,
+  canDrop: () => boolean,
+  onDrop: () => void,
+  style: {
+    dropIndicator: DropTargetContainerStyle,
+    dropArea: DropTargetContainerStyle,
+  },
+|}) {
+  const gdevelopTheme = React.useContext(GDevelopThemeContext);
+  return (
+    <DnDComponent canDrop={canDrop} drop={onDrop}>
+      {({ isOver, connectDropTarget, canDrop }) => {
+        return connectDropTarget(
+          <div>
+            {/* Drop area */}
+            <div
+              style={{
+                ...styles.dropArea,
+                ...style.dropArea,
+              }}
+            />
+            {/* Drop indicator */}
+            {canDrop && isOver && (
+              <div
+                style={{
+                  ...styles.dropIndicator,
+                  ...style.dropIndicator,
+                  borderColor: gdevelopTheme.dropIndicator.canDrop,
+                  outlineColor: gdevelopTheme.dropIndicator.border,
+                }}
+              />
+            )}
+          </div>
+        );
+      }}
+    </DnDComponent>
+  );
+}
+
+export function DropContainer({
+  node,
+  draggedNode,
+  DnDComponent,
+  onDrop,
+  activateTargets,
+  windowWidth,
+  draggedNodeHeight,
+}: {|
+  node: SortableTreeNode,
+  draggedNode: ?SortableTreeNode,
+  DnDComponent: any,
+  onDrop: (
+    moveFunction: ({
+      targetNode: SortableTreeNode,
+      node: SortableTreeNode,
+    }) => void,
+    node: SortableTreeNode
+  ) => void,
+  activateTargets: boolean,
+  windowWidth: WidthType,
+  draggedNodeHeight: number,
+|}) {
+  const isDraggedNodeNodesOnlyChild =
+    node.children.length === 1 &&
+    !!draggedNode &&
+    node.children[0].key === draggedNode.key;
+  const isDraggedNodeSameDepthAndBelow =
+    !!draggedNode && isSameDepthAndBelow(node, draggedNode);
+  // We want to allow dropping below if the event has no children OR if the only
+  // child of the event is the dragged one.
+  const canDropBelow =
+    node.event && (node.children.length === 0 || isDraggedNodeNodesOnlyChild);
+  const canHaveSubevents = node.event && node.event.canHaveSubEvents();
```

Comment:
nit:
```suggestion
  const canHaveSubevents = !!node.event && node.event.canHaveSubEvents();
```

to force this variable to be a boolean